### PR TITLE
Deleted keywords "virtual" where Keywords "override" is defined.

### PR DESCRIPTION
**Pull request by ZigRazor**
Following Guidelines rule C128 (ISSUE #17 Code style)

### CODE REVIEW

File: NeoMathEngine/src/CPU/CpuMathEngine.h [hunk 1]
```diff
@@ -36,34 +36,34 @@ class CCpuMathEngine : public IMathEngine, public IRawMemoryManager {
...
+	void VectorFill(const CFloatHandle& result, int vectorSize, const CConstFloatHandle& value) override;
+	void VectorFill(const CIntHandle& result, int vectorSize, const CConstIntHandle& value) override;
+	void VectorFillBernoulli( const CFloatHandle& result, float p, int vectorSize, float value, int seed ) override;
+	void FilterSmallValues( const CFloatHandle& data, int dataSize, float threshold ) override;
+	void VectorCopy(const CFloatHandle& first, const CConstFloatHandle& second, int vectorSize) override;
+	void VectorCopy(const CIntHandle& first, const CConstIntHandle& second, int vectorSize) override;
+	void VectorSum(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
+	void VectorSumAdd(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
+	void VectorNegSum(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
 	virtual void VectorEqual( const CConstIntHandle& firstHandle, const CConstIntHandle& secondHandle,
```
Comment (FedyuninV) [inline NeoMathEngine/src/CPU/CpuMathEngine.h:67]: In this file there're many multi-line function declarations. I think `virtual` should be removed there too.

But this file doesn't contain interfaces, that's why you can simply add `override` to destructor and remove `virtual` from the whole file.

File: NeoMathEngine/src/GPU/CUDA/CudaMathEngine.h [hunk 2]
```diff
@@ -47,34 +47,34 @@ class CCudaMathEngine : public IMathEngine, public IRawMemoryManager {
...
+	void VectorFill(const CFloatHandle& result, int vectorSize, const CConstFloatHandle& value) override;
+	void VectorFill(const CIntHandle& result, int vectorSize, const CConstIntHandle& value) override;
+	void VectorFillBernoulli( const CFloatHandle& result, float p, int vectorSize, float value, int seed ) override;
+	void FilterSmallValues( const CFloatHandle& data, int dataSize, float threshold ) override;
+	void VectorCopy(const CFloatHandle& first, const CConstFloatHandle& second, int vectorSize) override;
+	void VectorCopy(const CIntHandle& first, const CConstIntHandle& second, int vectorSize) override;
+	void VectorSum(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
+	void VectorSumAdd(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
+	void VectorNegSum(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
 	virtual void VectorEqual( const CConstIntHandle& firstHandle, const CConstIntHandle& secondHandle,
```
Comment (FedyuninV) [inline NeoMathEngine/src/GPU/CUDA/CudaMathEngine.h:78]: In this file there're many multi-line function declarations. I think `virtual` should be removed there too.

But this file doesn't contain interfaces, that's why you can simply add `override` to destructor and remove `virtual` from the whole file.

File: NeoMathEngine/src/GPU/Vulkan/VulkanMathEngine.h [hunk 3]
```diff
@@ -56,34 +56,34 @@ class CVulkanMathEngine : public IMathEngine, public IRawMemoryManager {
...
+	void VectorFill(const CFloatHandle& result, int vectorSize, const CConstFloatHandle& value) override;
+	void VectorFill(const CIntHandle& result, int vectorSize, const CConstIntHandle& value) override;
+	void VectorFillBernoulli( const CFloatHandle& result, float p, int vectorSize, float value, int seed ) override;
+	void FilterSmallValues( const CFloatHandle& data, int dataSize, float threshold ) override;
+	void VectorCopy(const CFloatHandle& first, const CConstFloatHandle& second, int vectorSize) override;
+	void VectorCopy(const CIntHandle& first, const CConstIntHandle& second, int vectorSize) override;
+	void VectorSum(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
+	void VectorSumAdd(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
+	void VectorNegSum(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
 	virtual void VectorEqual( const CConstIntHandle& firstHandle, const CConstIntHandle& secondHandle,
```
Comment (FedyuninV) [inline NeoMathEngine/src/GPU/Vulkan/VulkanMathEngine.h]: In this file there're many multi-line function declarations. I think `virtual` should be removed there too.

But this file doesn't contain interfaces, that's why you can simply add `override` to destructor and remove `virtual` from the whole file.

File: NeoMathEngine/src/CPU/CpuMathEngine.h (hunk 1 above)
Comment (FedyuninV) [inline NeoMathEngine/src/CPU/CpuMathEngine.h]: In this file there're many multi-line function declarations. I think `virtual` should be removed there too.

But this file doesn't contain interfaces, that's why you can simply add `override` to destructor and remove `virtual` from the whole file.

File: NeoMathEngine/src/GPU/CUDA/CudaMathEngine.h (hunk 2 above)
Comment (FedyuninV) [inline NeoMathEngine/src/GPU/CUDA/CudaMathEngine.h]: In this file there're many multi-line function declarations. I think `virtual` should be removed there too.

But this file doesn't contain interfaces, that's why you can simply add `override` to destructor and remove `virtual` from the whole file.

File: NeoMathEngine/src/GPU/Metal/MetalMathEngine.h [hunk 4]
```diff
@@ -44,34 +44,34 @@ class CMetalMathEngine : public IMathEngine, public IRawMemoryManager {
...
+	void VectorFill(const CFloatHandle& result, int vectorSize, const CConstFloatHandle& value) override;
+	void VectorFill(const CIntHandle& result, int vectorSize, const CConstIntHandle& value) override;
+	void VectorFillBernoulli( const CFloatHandle& result, float p, int vectorSize, float value, int seed ) override;
+	void FilterSmallValues( const CFloatHandle& data, int dataSize, float threshold ) override;
+	void VectorCopy(const CFloatHandle& first, const CConstFloatHandle& second, int vectorSize) override;
+	void VectorCopy(const CIntHandle& first, const CConstIntHandle& second, int vectorSize) override;
+	void VectorSum(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
+	void VectorSumAdd(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
+	void VectorNegSum(const CConstFloatHandle& firstHandle, int vectorSize, const CFloatHandle& resultHandle) override;
 	virtual void VectorEqual( const CConstIntHandle& firstHandle, const CConstIntHandle& secondHandle,
```
Comment (FedyuninV) [inline NeoMathEngine/src/GPU/Metal/MetalMathEngine.h]: In this file there're many multi-line function declarations. I think `virtual` should be removed there too.

But this file doesn't contain interfaces, that's why you can simply add `override` to destructor and remove `virtual` from the whole file.

**Review by FedyuninV — CHANGES_REQUESTED** [review]
Apologies for a delay in review.

Please, remove the rest of the `virtual`'s in Math Engines and then PR will be merged.

**Comment by ZigRazor** [comment]
I solved the requested change, so i request new review

**Review by FedyuninV — APPROVED** [review]
Thanks for contribution!

**Pull request merged**